server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java

```diff
@@ -318,30 +322,29 @@ Settings builtAnalyzerSettings() {
         return analyzerSettings;
     }
 
-    private static void applyAndValidateStorageSettings(Map<String, Object> mapping,
-                                                        AnalyzedColumnDefinition<Object> definition) {
+    public static boolean docValuesSpecifiedAndDisabled(AnalyzedColumnDefinition<Object> definition) {
```

Comment:
I updated it to return `doc_values` flag as this code can be used now also for building Reference. But as discussed in [], this is actually redundant and if I don't use Reference in `AddColumnRequest` I will revert this change.